aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -0,0 +1,208 @@
+#ifndef USE_MAGMA
+AT_ERROR("gesv: MAGMA library not found in "
+    "compilation. Please rebuild with MAGMA.");
+#else
+  auto A_data = A.data<scalar_t>();
+  auto b_data = b.data<scalar_t>();
+  auto A_mat_stride = matrixStride(A);
+  auto b_mat_stride = matrixStride(b);
+
+  magma_int_t batch_size = magma_int_cast(batchCount(A), "batchCount");
+  magma_int_t n = magma_int_cast(A.size(-2), "A.size(-2)");
+  magma_int_t nrhs = magma_int_cast(b.size(-1), "b.size(-1)");
+
+  magma_int_t* info_array;
+  magma_int_t* ipiv_data;
+  magma_int_t** ipiv_array;
+  scalar_t** A_array;
+  scalar_t** b_array;
+
+  ALLOCATE_ARRAY(info_array, magma_int_t, batch_size, b);
+  ALLOCATE_ARRAY(ipiv_data, magma_int_t, batch_size * n, b);
+  ALLOCATE_ARRAY(ipiv_array, magma_int_t*, batch_size, b);
+  ALLOCATE_ARRAY(A_array, scalar_t*, batch_size, b);
+  ALLOCATE_ARRAY(b_array, scalar_t*, batch_size, b);
+
+  // Set up the created arrays
+  for (int64_t i = 0; i < batch_size; i++) {
+    A_array[i] = &A_data[i * A_mat_stride];
+    b_array[i] = &b_data[i * b_mat_stride];
+    ipiv_array[i] = &ipiv_data[i * n];
+  }
+
+  magmaGesvBatched<scalar_t>(
+      n, nrhs, A_array, n, ipiv_array, b_array, n,
+      info_array, batch_size, MAGMAQueue(b.get_device()));
+
+  for (int64_t i = 0; i < batch_size; i++) {
+    infos[i] = info_array[i];
+  }
+#endif
+}
+
+GENERATE_LINALG_HELPER_2_ARGS(gesv, self, A, cuda)
+
+template <typename scalar_t>
+static void apply_inverse(Tensor &self, Tensor &self_inv, std::vector<int64_t>& infos) {
+#ifndef USE_MAGMA
+AT_ERROR("inverse: MAGMA library not found in "
+    "compilation. Please rebuild with MAGMA.");
+#else
+  auto self_data = self.data<scalar_t>();
+  auto self_mat_stride = matrixStride(self);
+  auto self_inv_data = self_inv.data<scalar_t>();
+  auto self_inv_mat_stride = matrixStride(self_inv);
+
+  magma_int_t batch_size = magma_int_cast(batchCount(self), "batchCount");
+  magma_int_t n = magma_int_cast(self.size(-2), "self.size(-2)");
+
+  magma_int_t* info_array;
+  magma_int_t* ipiv_data;
+  magma_int_t** ipiv_array;
+  scalar_t** self_array;
+  scalar_t** self_inv_array;
+
+  ALLOCATE_ARRAY(info_array, magma_int_t, batch_size, self);
+  ALLOCATE_ARRAY(ipiv_data, magma_int_t, batch_size * n, self);
+  ALLOCATE_ARRAY(ipiv_array, magma_int_t*, batch_size, self);
+  ALLOCATE_ARRAY(self_array, scalar_t*, batch_size, self);
+  ALLOCATE_ARRAY(self_inv_array, scalar_t*, batch_size, self_inv);
+
+  // Set up the created arrays
+  for (int64_t i = 0; i < batch_size; i++) {
+    self_array[i] = &self_data[i * self_mat_stride];
+    self_inv_array[i] = &self_inv_data[i * self_inv_mat_stride];
+    ipiv_array[i] = &ipiv_data[i * n];
+  }
+
+  magmaGetrfBatched<scalar_t>(
+    n, n, self_array, n, ipiv_array, info_array,
+    batch_size, MAGMAQueue(self.get_device()));
+
+  magmaGetriBatched<scalar_t>(
+    n, self_array, n, ipiv_array, self_inv_array,
+    n, info_array, batch_size, MAGMAQueue(self.get_device()));
```

Comment:
You should share this MAGMAQueue between getrf and getri. Something like the following:

```
MAGAMAQueue magma_queue(self.get_device());

  magmaGetrfBatched<scalar_t>(
    n, n, self_array, n, ipiv_array, info_array,
    batch_size, magma_queue);
   magmaGetriBatched<scalar_t>(
    n, self_array, n, ipiv_array, self_inv_array,
    n, info_array, batch_size, magma_queue);

```

Reply:
Sure, thank you.